src/pages/HomePage/usePagination.ts

```diff
@@ -0,0 +1,38 @@
+import { ChangeEvent, useState } from 'react';
+
+import { RoomDataProps } from './DummyData';
+
+interface usePaginationProps {
+  filteredData: RoomDataProps[];
+  itemPerPage?: number;
+}
+
+const usePagination = ({
+  filteredData,
+  itemPerPage = 4, // 한 페이지당 표시될 방의 갯수
+}: usePaginationProps) => {
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const lastItem = currentPage * itemPerPage;
+  const firstItem = lastItem - itemPerPage;
+  // 전체 페이지 수 계산
+  const pageCount =
+    filteredData.length % itemPerPage === 0
+      ? filteredData.length / itemPerPage
+      : Math.floor(filteredData.length / itemPerPage) + 1;
```

Comment:
Math.ceil을 이용하면 이렇게 줄여볼 수 있을 것 같아요! 👀

```suggestion
  const pageCount = Math.ceil(filteredData.length / itemPerPage);
```

Reply:
좋은 의견 감사드립니다 👍 

ceil() 같은 경우, 1 페이지에 대한 처리가 불가한 점이 있어, 현재와 같이 코드를 작성하였습니다.

ex ) 
화면에 출력할 방들의 갯수: 2개, 
한 화면에 최대로 출력될 방들의 갯수: 4개

이때, ceil() 를 사용하면 1페이지가 아니라 0페이지가 화면에 출력되기에, 해당 예외를 처리한 코드를 작성하였습니다.

Reply:
아하..! 내림 방식만 생각했었는데, 올림 ceil 함수를 사용하면 위 코드도 간단히 나타낼 수 있군요!
감사합니다 👍